Common/util/utf8.h

```diff
@@ -106,6 +106,14 @@ inline size_t GetLength(const char *c)
     return len;
 }
 
+// Takes a utf8 string pointer and rolls it back one char, unless it hits the head of same string
+inline const char* BackOneChar(const char* c, const char* front)
+{
+    if(c <= front) return front;
```

Comment:
I think this if here is redundant, because `--c` is safe to do, and `c > front` is tested first in the loop condition.